### PR TITLE
Adds Header component

### LINKED ISSUE #3: Criar componente do header

### PR DESCRIPTION
**Pull request by RodrigoEC**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,6 +2,7 @@ PWD=$(shell pwd)
 NODE_IMAGE=node:20.12.2
 IMAGE_NAME=oca-frontend
 CONTAINER_NAME=oca-frontend-app
+PORT=3000
```
Comment (cilasmarques) [inline Makefile]: Define the container port and the host port 

```suggestion
HOST_PORT=80
CONTAINER_PORT=3000
```

File: Makefile [hunk 2]
```diff
@@ -11,13 +12,13 @@ run-prod:
 	npm run start
 
 docker-run-dev:
```
Comment (cilasmarques) [inline Makefile:15]: Map the container port to the host port (in our case the http port ~ 80)

```suggestion
docker-run-dev:
	docker run -p $(HOST_PORT):$(CONTAINER_PORT) --name $(IMAGE_NAME) -v node_modules -v $(PWD):/app $(IMAGE_NAME)
```

File: src/components/Header/Header.tsx [hunk 3]
```diff
@@ -1,45 +1,46 @@
...
+      href: "#mapa-e-visualizações",
+    },
+    {
+      name: "Nosso time",
+      href: "#nosso-time",
+    },
+    {
+      name: "Apoiadores",
+      href: "#apoiadores",
+    },
```
Comment (cilasmarques) [inline src/components/Header/Header.tsx]: Standardize hrefs - all in English or all in Portuguese.

File: src/components/MenuModal/MenuModal.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  }, []);
+
+  const switchRetract = useCallback(() => {
+    if (retracted) {
+      document.addEventListener("click", handleClickOutside);
+    } else {
+      document.removeEventListener("click", handleClickOutside);
+    }
+
+    console.log("oi", retracted)
```
Comment (cilasmarques) [inline src/components/MenuModal/MenuModal.tsx]: Remove console.log

File: src/components/MenuModal/MenuModal.tsx [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    setRetracted(previous => !previous);
+  }, [handleClickOutside, retracted]);
+
+  return (
+    <div {...props}>
+      <a onClick={switchRetract}>
+        <MenuImage src={MenuIcon} alt="" />
+      </a>
```
Comment (cilasmarques) [inline src/components/MenuModal/MenuModal.tsx]: Why not put the onclick directly on the image?
```suggestion
      <MenuImage onClick={switchRetract} src={MenuIcon} alt="" />
```

**Review by cilasmarques — CHANGES_REQUESTED** [review]

**Comment by cilasmarques** [comment]
Very Nice!

File: src/components/MenuModal/MenuModal.tsx (hunk 5 above)
Comment (RodrigoEC) [reply src/components/MenuModal/MenuModal.tsx]: Because the Next's Image component doesn't accept the onClick prop, we have to wrap it with another tag

Comment (RodrigoEC) [reply src/components/MenuModal/MenuModal.tsx]: I changed it from a tag to a div tag to be more semantically correct

**Review by cilasmarques — APPROVED** [review]

**Pull request merged**